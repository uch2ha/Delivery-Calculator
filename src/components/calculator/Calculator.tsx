import './Calculator.css';
import React from 'react';
import { RiMoneyEuroCircleLine } from 'react-icons/ri';

interface IProps {
  cartValue: number;
  distance: number;
  itemsAmount: number;
  userDate: string;
  deliveryPrice: number;
  handleInput: React.ChangeEventHandler<HTMLInputElement>;
  calculateDeliveryHandler: Function;
  handleRefresh: Function;
}

const Calculator: React.FC<IProps> = ({
  cartValue,
  distance,
  itemsAmount,
  userDate,
  deliveryPrice,
  handleInput,
  calculateDeliveryHandler,
  handleRefresh,
}) => {
  return (
    <>
      <div className='calculator-inputs'>
        <div>
          <span className='input-name'>Cart value</span>
        </div>
        <div>
          <span className='input-body'>
            <input
              type='text'
              id='cartValue'
              value={cartValue}
              onChange={handleInput}
              data-testid='cartValue'
            />
            <RiMoneyEuroCircleLine size={30} />
          </span>
        </div>
        <div>
          <span className='input-name'>Delivery distance</span>
        </div>
        <div>
          <span className='input-body delivery-icon'>
            <input
              type='text'
              id='distance'
              value={distance}
              onChange={handleInput}
              data-testid='distance'
            />
            m
          </span>
        </div>
        <div>
          <span className='input-name'>Amount of items</span>
        </div>
        <div>
          <span className='input-body'>
            <input
              type='text'
              id='itemsAmount'
              value={itemsAmount}
              onChange={handleInput}
              data-testid='itemsAmount'
            />
          </span>
        </div>
        <div>
          <span className='input-name'>Time</span>
        </div>
        <div>
          <span className='input-body'>
            <input
              className='date-input'
              id='date'
              type='datetime-local'
              value={userDate}
              onChange={handleInput}
              data-testid='date'
            />
          </span>
        </div>
      </div>
      <div className='calculator-btn'>
        <div
          className='btn-calculate btn'
          onClick={() => calculateDeliveryHandler()}
        >
          <span>Calculate delivery price</span>
        </div>
        <div className='btn-refresh btn' onClick={() => handleRefresh()}>
          <span>Refresh</span>
        </div>
      </div>
      <div className='calculator-result'>
        <div>
          <span>Delivery price</span>
          <div>
            <span data-testid='result'>= {deliveryPrice}</span>
            <RiMoneyEuroCircleLine size={30} />
          </div>
        </div>
      </div>
    </>
  );
};

export default Calculator;
