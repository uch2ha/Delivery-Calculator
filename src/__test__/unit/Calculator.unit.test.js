import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Calculator from '../../components/calculator/Calculator';

const mockHandleInput = jest.fn();
const mockCalculateDeliveryHandler = jest.fn();
const mockHandleRefresh = jest.fn();

describe('Calculator component', () => {
  let container;

  beforeEach(() => {
    container = render(
      <Calculator
        cartValue={0}
        distance={0}
        itemsAmount={0}
        userDate=''
        deliveryPrice={0}
        handleInput={mockHandleInput}
        calculateDeliveryHandler={mockCalculateDeliveryHandler}
        handleRefresh={mockHandleRefresh}
      />
    );
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    expect(container.baseElement).toMatchSnapshot();
  });

  it('should call handleInput when an input value changes', () => {
    const cartValueInput = container.getByTestId('cartValue');
    fireEvent.change(cartValueInput, { target: { value: '100' } });
    expect(mockHandleInput).toHaveBeenCalled();
  });

  it('should call calculateDeliveryHandler when calculate delivery price button is clicked', () => {
    const calculateBtn = container.getByText(/Calculate delivery price/i);
    fireEvent.click(calculateBtn);
    expect(mockCalculateDeliveryHandler).toHaveBeenCalled();
  });

  it('should call handleRefresh when refresh button is clicked', () => {
    const refreshBtn = container.getByText(/Refresh/i);
    fireEvent.click(refreshBtn);
    expect(mockHandleRefresh).toHaveBeenCalled();
  });
});
