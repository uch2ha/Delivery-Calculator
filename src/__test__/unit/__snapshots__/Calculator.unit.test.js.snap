// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component should render correctly 1`] = `
<body>
  <div>
    <div
      class="calculator-inputs"
    >
      <div>
        <span
          class="input-name"
        >
          Cart value
        </span>
      </div>
      <div>
        <span
          class="input-body"
        >
          <input
            data-testid="cartValue"
            id="cartValue"
            type="text"
            value="0"
          />
          <svg
            fill="currentColor"
            height="30"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zm0-2a8 8 0 1 0 0-16 8 8 0 0 0 0 16zm-1.95-9H15v2h-4.95a2.5 2.5 0 0 0 4.064 1.41l1.7 1.133A4.5 4.5 0 0 1 8.028 13H7v-2h1.027a4.5 4.5 0 0 1 7.788-2.543L14.114 9.59A2.5 2.5 0 0 0 10.05 11z"
              />
            </g>
          </svg>
        </span>
      </div>
      <div>
        <span
          class="input-name"
        >
          Delivery distance
        </span>
      </div>
      <div>
        <span
          class="input-body delivery-icon"
        >
          <input
            data-testid="distance"
            id="distance"
            type="text"
            value="0"
          />
          m
        </span>
      </div>
      <div>
        <span
          class="input-name"
        >
          Amount of items
        </span>
      </div>
      <div>
        <span
          class="input-body"
        >
          <input
            data-testid="itemsAmount"
            id="itemsAmount"
            type="text"
            value="0"
          />
        </span>
      </div>
      <div>
        <span
          class="input-name"
        >
          Time
        </span>
      </div>
      <div>
        <span
          class="input-body"
        >
          <input
            class="date-input"
            data-testid="date"
            id="date"
            type="datetime-local"
            value=""
          />
        </span>
      </div>
    </div>
    <div
      class="calculator-btn"
    >
      <div
        class="btn-calculate btn"
      >
        <span>
          Calculate delivery price
        </span>
      </div>
      <div
        class="btn-refresh btn"
      >
        <span>
          Refresh
        </span>
      </div>
    </div>
    <div
      class="calculator-result"
    >
      <div>
        <span>
          Delivery price
        </span>
        <div>
          <span
            data-testid="result"
          >
            = 
            0
          </span>
          <svg
            fill="currentColor"
            height="30"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zm0-2a8 8 0 1 0 0-16 8 8 0 0 0 0 16zm-1.95-9H15v2h-4.95a2.5 2.5 0 0 0 4.064 1.41l1.7 1.133A4.5 4.5 0 0 1 8.028 13H7v-2h1.027a4.5 4.5 0 0 1 7.788-2.543L14.114 9.59A2.5 2.5 0 0 0 10.05 11z"
              />
            </g>
          </svg>
        </div>
      </div>
    </div>
  </div>
</body>
`;
